src/dotnet/APIView/APIViewWeb/Managers/SamplesRevisionsManager.cs

```diff
@@ -87,6 +113,18 @@ public async Task<SamplesRevisionModel> UpsertSamplesRevisionsAsync(ClaimsPrinci
             return await UpsertSamplesRevisionsAsync(user, reviewId, sample, revisionTitle, FileName);
         }
 
+        public async Task UpdateSamplesRevisionAsync(ClaimsPrincipal user, string reviewId, string sampleRevisionId, string newContent = null, string newTitle = null)
+        {
+            var samplesRevision = await _samplesRevisionsRepository.GetSamplesRevisionAsync(reviewId, sampleRevisionId);
+            await AssertUsageSampleOwnerAsync(user, samplesRevision);
+            
+            samplesRevision.Title = newTitle;
+            var originalStream = new MemoryStream(Encoding.UTF8.GetBytes(newContent));
+
+            await _samplesRevisionsRepository.UpsertSamplesRevisionAsync(samplesRevision);
```

Comment:
Let's upload the file first and then save the metadata. Otherwise, if storage request fails for any reason, then we will end up having stale data in cosmos.